pkg/common/config/config.go

```diff
@@ -299,16 +300,21 @@ func validateConfig(ctx context.Context, cfg *Config) error {
 	if err != nil {
 		return err
 	}
-	if cfg.FeatureStatesConfig.Name == "" && cfg.FeatureStatesConfig.Namespace == "" {
-		cfg.FeatureStatesConfig.Name = DefaultFSSConfigMapName
-		if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
-			// If feature states config info is not provided in vsphere conf, use defaults for vanilla k8s cluster
-			log.Infof("No feature states config information is provided in the Config. Using default config map name: %s and namespace: %s", DefaultFSSConfigMapName, DefaultFSSConfigMapNamespaceVanillaK8s)
-			cfg.FeatureStatesConfig.Namespace = DefaultFSSConfigMapNamespaceVanillaK8s
-		} else if clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
-			// Feature states config info is not provided in vsphere conf in project pacific, use defaults for supervisor cluster
-			cfg.FeatureStatesConfig.Namespace = DefaultCSINamespace
-		}
+	// Validate FeatureStateConfig used in Supervisor cluster
```

Comment:
Do we want to validate that "Internal features states" cannot exist in the SV cluster and similarly "Feature states" cannot exist in Vanilla?

Reply:
I have separated them out, so we do not need validation anymore